components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1087,10 +1091,20 @@ private void deleteApplicationRole(String applicationName) {
     private ServiceProvider doAddApplication(ServiceProvider serviceProvider, String tenantDomain, String username)
             throws IdentityApplicationManagementException {
 
+        if (StringUtils.isBlank(serviceProvider.getApplicationName())) {
+            // check for required attributes.
+            throw new IdentityApplicationManagementException("Application Name is required");
+        }
+
+        ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
+        ServiceProvider savedSP = appDAO.getApplication(serviceProvider.getApplicationName(), tenantDomain);
```

Comment:
In order to check existence of a SP, getApplication() would not be the efficient way, since it'll populate whole service provider object. Better to create a separate method in DAO layer to check the existence(i.e. check in SP_APP table, whether an entry exists for the specified application name)

Reply:
Created an issue to track this improvement https://github.com/wso2/product-is/issues/3621